File: ja/code_snippets/api-metrics-query.rb
require 'rubygems'
require 'dogapi'

api_key = "<YOUR_API_KEY>"
application_key = "<YOUR_APP_KEY>"

dog = Dogapi::Client.new(api_key, application_key)

# Get points from the last hour
from = Time.now - 3600
to = Time.now
query = 'system.cpu.idle{*}by{host}'

dog.get_points(query, from, to)

File: content/en/api/metrics/code_snippets/api-metrics-query.py
from datadog import initialize, api
import time

options = {
    'api_key': '<DATADOG_API_KEY>',
    'app_key': '<DATADOG_APPLICATION_KEY>'
}

initialize(**options)

now = int(time.time())

query = 'system.cpu.idle{*}by{host}'
print(api.Metric.query(start=now - 3600, end=now, query=query))
